### PR TITLE
Annotate umis

### PR DESCRIPTION
**Pull request by FelixMoelder**
Pipeline expansion for annotation of molecular barcodes (UMIs).
UMIs will be annotated right after read mapping.
In case of annotation a the `BARCODE`-parameter will be set when executing `picardtools MarkDuplicates`.
This allows building consensus reads by UMI as duplicates are now grouped by UMI.

### CODE REVIEW

File: workflow/rules/plugins.smk
Comment (FelixMoelder) [inline workflow/rules/plugins.smk:7]: This url comes from a dynamic google link. Let's how this never changes...

File: workflow/rules/ref.smk
Comment (FelixMoelder) [inline workflow/rules/ref.smk:104]: This is always part of the VEP plugins. But we need to add this to the input of `annotate_variants` for correct handling of the `LoFtool`-plugin.

File: workflow/rules/annotation.smk
Comment (FelixMoelder) [inline workflow/rules/annotation.smk:27]: These are case sensitive as the VEP snakemake-wrapper is looking for plugin aux-files in the input by looking for the plugin-name.
Alternatively, we could set this to lowercase in the wrapper but this might lead to confusion for users.
Any suggestions?

**Comment by FelixMoelder** [comment]
This PR has already been tested on real data.
Before merging we should probably wait for a new varlociraptor release.

File: config/config.yaml
Comment (johanneskoester) [inline config/config.yaml:122]: Please also add the revel setting to this template config (commented out).

**Review by johanneskoester — CHANGES_REQUESTED** [review]
Nice and clean! Just one comment below.

File: config/config.yaml
Comment (FelixMoelder) [reply config/config.yaml:122]: I guess you mean just adding `REVEL` as a plugin in the config?
This is done now.

File: workflow/rules/common.smk
Comment (FelixMoelder) [inline workflow/rules/common.smk:500]: I removed this as this seems to be dead code. At least I could not find where this function is called.

File: .test/config-sra/config.yaml
Comment (johanneskoester) [inline .test/config-sra/config.yaml:52]: For convenience, it should still be possible to put just a string here without a list. Can you keep it that way for one of the tests?

File: workflow/rules/common.smk
Comment (johanneskoester) [inline workflow/rules/common.smk:544]: Allow filter to be just a string here (for convenience and compatibility).

File: workflow/schemas/units.schema.yaml
Comment (johanneskoester) [inline workflow/schemas/units.schema.yaml:28]: Shouldn't there also be an update to config.schema.yaml, for the filter changes?

**Review by johanneskoester — CHANGES_REQUESTED** [review]

File: .test/config-sra/config.yaml
Comment (FelixMoelder) [reply .test/config-sra/config.yaml:52]: Good point. I just reversed one test to cover this.

File: workflow/rules/common.smk
Comment (FelixMoelder) [reply workflow/rules/common.smk:544]: This should work now.

**Review by johanneskoester — APPROVED** [review]

**Pull request merged**
